src/main/java/org/traccar/protocol/Xexun2ProtocolDecoder.java

```diff
@@ -99,12 +100,12 @@ protected Object decode(
             return null;
         }
 
-        sendResponse(channel, type, index, imei);
-
         buf.readUnsignedShort(); // attributes
         buf.readUnsignedShort(); // checksum
 
         if (type == MSG_POSITION) {
+            sendResponse(channel, type, index, imei);
```

Comment:
Why is this moved?

Reply:
When sending a command (type = 0x07) the device will acknowledge the command with a message type = 0x07.

If decoder responded to the command ack with another ack, the device would interpret that as another command and create a loop.

Decoder should only ack (`sendResponse()`) when type = 0x14

Reply:
Shouldn't we send to `type != 0x07`?

Reply:
I don't think it makes a difference. As far as I know only 0x07 and 0x14 are used for message id. None other are mentioned in documentation.

I'm not sure the server should blindly acknowledge other messages? Although I realise it might be a breaking change if devices do send other message id, so perhaps for backwards compatibility sending a response when type != 0x07 is safer?

Your call.

Reply:
Let's do `type != 0x07`. This is more in line with other protocols, so I think that's probably be right if they extend the protocol.